sw/device/tests/alert_handler_lpg_reset_toggle.c

```diff
@@ -0,0 +1,548 @@
+            "ping_timrout_fail = 1 for peripheral[%d]",
+            peri_idx);
+
+      // Release the reset
+      CHECK_DIF_OK(dif_rstmgr_software_reset(&rstmgr,
+                                             kPeripherals[peri_idx].reset_index,
+                                             kDifRstmgrSoftwareResetRelease));
+
+      // Increment the test_step counter
+      flash_ctrl_testutils_counter_increment(&flash_ctrl, kCounterTestSteps);
+      // Request system reset to unlock the alert handler config and test the
+      // next peripheral
+      CHECK_DIF_OK(dif_rstmgr_software_device_reset(&rstmgr));
+    } else {
+      // For the simulation, only increment the test_step counter
+      // to procced to the next phase without resetting the device
+      test_step_cnt++;
+    }
+  }
+
+  /* TEST PHASE #1: TEST THE REGULAR CONFIG
+
+    The timeout value is set to 256
+    None of the alerts should trigger ping_timeout_alert
+    TEST:
+    enable_alerts (all_peripherals, timeout=256)
+    for_each_peripheral
+      1- wait for random time
+      2- Hold the IP's reset
+      3- Wait long enough
+      4- Release the IP's reset
+      5- Confirm that ping_timeout has not been triggered
+  */
+
+  // Enable and lock the all peripherals' alerts at the beginning
+  // of the PHASE #1
+  if (test_step_cnt == 1 * ARRAYSIZE(kPeripherals)) {
+    alert_handler_config_peripherals(
+        /*num_peripherals*/ ARRAYSIZE(kPeripherals),
+        /* *first_peripheral*/ &kPeripherals[0], /*ping_timeout*/ 256);
+  }
+
+  while (test_step_cnt < 2 * ARRAYSIZE(kPeripherals)) {
+    // Read the test_step_cnt and compute the test phase
+    // amd the peripheral ID to test
+    test_phase = test_step_cnt / ARRAYSIZE(kPeripherals);
+    peri_idx = test_step_cnt - (test_phase)*ARRAYSIZE(kPeripherals);
+
+    // Wait for a random time <= 1024us
+    get_rand_words(&ibex, /*number of words*/ 1, &rnd_wait_time,
+                   /*max*/ 1 << 10);
+    busy_spin_micros(rnd_wait_time);
+
+    // Hold the reset of the peripheral
+    CHECK_DIF_OK(dif_rstmgr_software_reset(&rstmgr,
+                                           kPeripherals[peri_idx].reset_index,
+                                           kDifRstmgrSoftwareResetHold));
+
+    // wait enough until the alert handler pings the peripheral
+    wait_enough_for_alert_ping();
+
+    // Check the alert status
+    CHECK_DIF_OK(dif_alert_handler_local_alert_is_cause(
+        &alert_handler, kDifAlertHandlerLocalAlertAlertPingFail, &is_cause));
+    CHECK(!is_cause, "Expected response: No alert_ping_fail! but we got %d",
+          is_cause);
+
+    // Release the reset of the peripheral
+    CHECK_DIF_OK(dif_rstmgr_software_reset(&rstmgr,
+                                           kPeripherals[peri_idx].reset_index,
+                                           kDifRstmgrSoftwareResetRelease));
+
+    // Increment the test counter
+    test_step_cnt++;
+  }
+
```

Comment:
Would it make sense to check at this point that none of the local alerts have been set in the alert handler? Just to be sure?

Reply:
I was just thinking that since this test (and the other LPG tests) provide good stimulus, it may make sense to check everything else is behaving as expected. this is similar to the IRQ check that we added above that makes sure we do not get any spurious ones.

anyways, we can add this check in a follow up.